app/src/main/kotlin/com/getstrm/pace/processing_platforms/ProcessingPlatformViewGenerator.kt

```diff
@@ -161,6 +139,7 @@ abstract class ProcessingPlatformViewGenerator(
         if (retention.conditionsList.size == 1) {
             // If there is only one filter it should be the only option
             // create retention sql
+            return condition(retention.conditionsList.first().toRetentionCondition(retention.field))
```

Comment:
why this? It seems to have little to do with the aggregation.

Reply:
Mentioned this in the first comment above https://github.com/getstrm/pace/pull/116#issuecomment-1852417282

Reply:
which bug? Is  there a bug description?

Reply:
Incomplete code. There was no correct handling of a retention filter with only one condition. Added it here and added a test